homeassistant/components/mqtt/discovery.py

```diff
@@ -236,16 +236,20 @@ async def discovery_done(_):
                         # pylint: disable-next=import-outside-toplevel
                         from . import device_automation
 
-                        await device_automation.async_setup_entry(hass, config_entry)
+                        hass.async_add_job(
+                            device_automation.async_setup_entry(hass, config_entry)
+                        )
```

Comment:
We should await here, otherwise the discovery signal is racing against the config entry setup